offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/products/ProductsDbConnector.groovy

```diff
@@ -220,11 +224,87 @@ class ProductsDbConnector {
     return productItems
   }
 
+  /**
+   * A product is archived by setting it inactive
+   * @param product The product that needs to be archived
+   * @since 1.0.0
+   * @throws life.qbic.business.exceptions.DatabaseQueryException
+   */
+  @Override
+  void archive(Product product) throws DatabaseQueryException {
+    Connection connection = provider.connect()
+
+    connection.withCloseable {
+      def statement = connection.prepareStatement(Queries.ARCHIVE_PRODUCT)
+      statement.setString(1, product.productId.toString())
+      statement.execute()
+    }
+  }
+
+  /**
+   * Fetches a product from the database
+   * @param productId The product id of the product to be fetched
+   * @return returns an optional that contains the product if it has been found
+   * @since 1.0.0
+   * @throws life.qbic.business.exceptions.DatabaseQueryException is thrown when any technical interaction with the data source fails
+   */
+  @Override
+  Optional<Product> fetch(ProductId productId) throws DatabaseQueryException {
+    Connection connection = provider.connect()
+    String query = Queries.SELECT_ALL_PRODUCTS + " WHERE productId=?"
+    Optional<Product> product = Optional.empty()
+
+    connection.withCloseable {
+      PreparedStatement preparedStatement = it.prepareStatement(query)
+      preparedStatement.setString(1, productId.identifier.toString())
+      ResultSet result = preparedStatement.executeQuery()
+
+      while (result.next()) {
+        product = Optional.of(rowResultToProduct(SqlExtensions.toRowResult(result)))
+      }
+    }
+    return product
+  }
+
+  /**
+   * Stores a product in the database
+   * @param product The product that needs to be stored
+   * @since 1.0.0
+   * @throws DatabaseQueryException if any technical interaction with the data source fails
+   * @throws ProductExistsException if the product already exists in the data source
+   */
+  @Override
+  void store(Product product) throws DatabaseQueryException, ProductExistsException {
+    Connection connection = provider.connect()
```

Comment:
There seems to be no test for a duplicate product entry aka the ProductExistsException case

Reply:
this is correct